web/src/pages/people/index.tsx

```diff
@@ -44,6 +54,33 @@ export default function PeoplePage() {
   const isAdmin = userData?.roles.includes("admin");
   const isIAO = userData?.roles.includes("information-asset-owner");
   const isTreOpsStaff = userData?.roles.includes("tre-ops-staff");
+  const canSearch = isTreOpsStaff || isAdmin;
+
+  const handleUserSearch = async (query: string) => {
+    const regex = /^\w[\w.\s0-9@]+\w$/;
+    const isValid = new RegExp(regex).test(query);
+    console.log(query, isValid);
+
+    if (!isValid) {
+      setSearchErrorMessage("invalid query; note that there's a minimum of 3 characters required to perform search");
+      setUsers(originalUsers);
```

Comment:
I'm not entirely sure about setting the original users on an invalid query. Typical users (who are not the dev's) won't know that this is actually setting the originally fetched users and not some other set of filtered users. It's obvious with just a few users in the list, but as the list of users grows it may become confusing.

Perhaps it would be better to not return any users for an invalid query so that it is obvious that nobody was returned? I think instinctively people will modify their search term if something went wrong, or delete it to return to the default set of fetched users.

Happy to hear other thoughts though.

Reply:
If there are no results, it _should_ say "no results for <query>". Possibly we've changed something today that's made that not so. if it's invalid - len 0, doesn't match the regex, then they get an error

Reply:
<img width="839" height="140" alt="image" src="https://github.com/user-attachments/assets/bf354bd9-6302-46e6-8bcb-8257c924930f" />
<img width="816" height="135" alt="image" src="https://github.com/user-attachments/assets/b4097d64-3905-4fe0-880f-c170fd3300f6" />

Reply:
Yes but if it hits an invalid regex the logic currently does: `setUsers(originalUsers);`. I'm suggesting that we don't do that and instead return no users for an invalid query.

But I could be wrong about that. I don't feel too strongly.

Reply:
so only if the query is empty again would we return the list of users?

Reply:
Alternatively, if it hits an invalid query the users simply are not reset and it remains as it was for the most recent query?